### PR TITLE
Migrate audio session manager to `NuguClientKit`

### PR DESCRIPTION
**Pull request by jin-marino**
## Check before PR

- [x] PR Title
- [x] Link issue or no issue to link
- [ ] README.md
- [x] Code-level documentation

## Version

- [ ] Update major
- [ ] Update minor
- [x] Update patch

## Need when updating version

- [ ] NUGU Developers
- [ ] Github Wiki

### CODE REVIEW

File: NuguClientKit/Sources/Audio/AudioSessionManager.swift [hunk 1]
```diff
@@ -19,26 +19,44 @@
 //
 
 import AVFoundation
-import UIKit
 
 import NuguAgents
 
-final class NuguAudioSessionManager {
-    static let shared = NuguAudioSessionManager()
+final public class AudioSessionManager: AudioSessionManageable {
```
Comment (Minddol) [inline NuguClientKit/Sources/Audio/AudioSessionManager.swift:25]: private let 으로 하면 될 것 같은데요?

Comment (jin-marino) [reply NuguClientKit/Sources/Audio/AudioSessionManager.swift:25]: 반영하였습니다~

File: NuguCore/Sources/StreamData/StreamDataRouter.swift [hunk 2]
```diff
@@ -27,17 +27,16 @@ import RxSwift
 public class StreamDataRouter: StreamDataRoutable {
     private let notificationQueue = DispatchQueue(label: "com.sktelecom.romaine.stream_data_router_notificaiton_queue")
     
-    private let nuguApiProvider = NuguApiProvider()
+    private lazy var nuguApiProvider = NuguApiProvider()
     private let directiveSequencer: DirectiveSequenceable
     @Atomic private var eventSenders = [String: EventSender]()
     @Atomic private var eventDisposables = [String: Disposable]()
-    private var serverInitiatedDirectiveRecever: ServerSideEventReceiver
+    private lazy var serverInitiatedDirectiveRecever = ServerSideEventReceiver(apiProvider: nuguApiProvider)
```
[the remark this answers is not in the document]
Comment (Minddol) [reply NuguCore/Sources/StreamData/StreamDataRouter.swift]: 수정했습니다~

File: NuguClientKit/Sources/Client/NuguClient.swift [hunk 3]
```diff
@@ -211,6 +211,8 @@ public class NuguClient {
             streamDataRouter: streamDataRouter,
             dialogStateAggregator: dialogStateAggregator
         )
+        
+        keywordDetector = KeywordDetector(contextManager: contextManager)
```
[the remark this answers is not in the document]
Comment (Minddol) [reply NuguClientKit/Sources/Client/NuguClient.swift]: 수정했습니다~

**Review by yHoonyK — APPROVED** [review]

**Review by Minddol — APPROVED** [review]

**Review by childc — APPROVED** [review]

**Pull request merged**